src/main/scala/com/exasol/cloudetl/kafka/KafkaConsumerProperties.scala

```diff
@@ -211,6 +212,39 @@ class KafkaConsumerProperties(private val properties: Map[String, String])
     props.toMap.asInstanceOf[Map[String, AnyRef]].asJava
   }
 
+  /**
+   * Returns a new [[KafkaConsumerProperties]] that merges the key-value pairs
+   * parsed from user provided Exasol named connection object.
+   */
+  final def mergeWithConnectionObject(exaMetadata: ExaMetadata): KafkaConsumerProperties = {
+    validateConnectionObject()
+    val connectionParsedMap =
+      parseConnectionInfo(SSL_KEYSTORE_LOCATION.userPropertyName, Option(exaMetadata))
+    val newProperties = properties ++ connectionParsedMap
+    new KafkaConsumerProperties(newProperties)
+  }
+
+  private[this] def validateConnectionObject(): Unit = {
+    val connectionProperties = List(
+      "BOOTSTRAP_SERVERS",
+      "SCHEMA_REGISTRY_URL",
+      "SECURITY_PROTOCOL",
+      "SSL_ENABLED",
+      "SSL_KEYSTORE_LOCATION",
+      "SSL_KEYSTORE_PASSWORD",
+      "SSL_KEY_PASSWORD",
+      "SSL_TRUSTSTORE_LOCATION",
+      "SSL_TRUSTSTORE_PASSWORD"
+    )
```

Comment:
Could you please limit these parameter to only `*_PASSWORD` parameters?

In general, the location can only be BucketFS bucket, so maybe it is okay that they are listed in the logs.

Reply:
Please also update the tests so that only passwords are provided with connection object.

Reply:
Hmm ok, I was thinking more from a connection management viewpoint.. eg. the connection object should completely encapsulate all of the connection details. If you're using a connection, it should have all of the details about the Kafka servers and Schema Registry, so that you can concern yourself only with topic etc in the IMPORT statement.

Also, if the connection object provides passwords, I wouldn't want to allow those to be "re-used" for any other certificate; eg. providing a different location, but using the password from the connection object. That could be a security issue if you by accident or policy maintain the same passwords for different certificates.

Reply:
Connection object is usually used to hide credentials from logs. 

But, I agree with you second point. Please keep the location information together with passwords in the connection object. I remember similar situation from other project authentication where we kept them together.

Reply:
I guess SSL_ENABLED doesn't *need* to be in that list, but on the other hand:

If you have a poorly configured Kafka cluster that allows non-SSL connections somehow (perhaps because the Exasol server sits on a trusted network), but as a DBA you *want* everyone to use SSL + certificate authentication to avoid plaintext data over the wire (and for traceability / auditing purposes on the Kafka end), I'm not sure it's good if a user can override it.

Reply:
Yes. It also depends on the Kafka setup. If it is setup with SSL authentication, users should be able to skip secure authentication. Not ideal, but provides an option for both situations.

Usually, we can only advise to use secure authentication but not force :)

Reply:
Yeah, it just sort of comes back to connection encapsulation (because the user can anyway use IMPORT without any connection object at all I guess). Eg. if I create a connection intended to be used with SSL enabled, it probably won't even work without SSL enabled anyway, and vice versa, so it doesn't make a lot of sense to override it outside of the connection object :)

Reply:
It would be nice by the way if the TO option of CREATE CONNECTION could be mapped to BOOTSTRAP_SERVERS.. :innocent:

Reply:
That is even better :)